java/drivers/driver-hazelcast4plus/src/main/java/com/hazelcast/simulator/tests/cp/CPMapTest.java

```diff
@@ -73,58 +103,119 @@ private String[] createCpGroupNames() {
     public void prepare() {
         for (CPMap<Integer, String> mapReference : mapReferences) {
             for (int key = 0; key < keys; key++) {
-                mapReference.set(key, v);
+                mapReference.set(key, INITIAL_VALUE);
             }
         }
     }
 
     @TimeStep(prob = 1)
     public void set(ThreadState state) {
-        state.randomMap().set(state.randomKey(), v);
+        state.randomMap().set(state.randomKey(), state.randomValue());
+        state.operationCounter.setCount.incrementAndGet();
     }
 
     @TimeStep(prob = 0)
-    public String put(ThreadState state) {
-        return state.randomMap().put(state.randomKey(), v);
+    public void put(ThreadState state) {
+        state.randomMap().put(state.randomKey(), state.randomValue());
+        state.operationCounter.putCount.incrementAndGet();
     }
 
     @TimeStep(prob = 0)
-    public String get(ThreadState state) {
-        return state.randomMap().get(state.randomKey());
+    public void putIfAbsent(ThreadState state) {
+        state.randomMap().putIfAbsent(state.randomKey(), state.randomValue());
+        state.operationCounter.putIfAbsentCount.incrementAndGet();
+    }
+
+    @TimeStep(prob = 0)
+    public void get(ThreadState state) {
+        state.randomMap().get(state.randomKey());
+        state.operationCounter.getCount.incrementAndGet();
     }
 
     // 'remove' and 'delete' other than their first invocation pointless -- we're just timing the logic that underpins the
     // retrieval of no value.
 
     @TimeStep(prob = 0)
-    public String remove(ThreadState state) {
-        return state.randomMap().remove(state.randomKey());
+    public void remove(ThreadState state) {
+        state.randomMap().remove(state.randomKey());
+        state.operationCounter.removeCount.incrementAndGet();
     }
 
     @TimeStep(prob = 0)
     public void delete(ThreadState state) {
         state.randomMap().delete(state.randomKey());
+        state.operationCounter.deleteCount.incrementAndGet();
     }
 
     @TimeStep(prob = 0)
-    public boolean cas(ThreadState state) {
-        // 'v' is always associated with 'k'
-        return state.randomMap().compareAndSet(state.randomKey(), v, v);
+    public void cas(ThreadState state) {
+        CPMap<Integer, String> randomMap = state.randomMap();
+        Integer key = state.randomKey();
+        String expectedValue = randomMap.get(key);
+        if (expectedValue != null) {
+            randomMap.compareAndSet(key, expectedValue, state.randomValue());
+            state.operationCounter.casCount.incrementAndGet();
+        }
     }
 
     @TimeStep(prob = 0)
     public void setThenDelete(ThreadState state) {
         CPMap<Integer, String> map = state.randomMap();
         int key = state.randomKey();
-        map.set(key, v);
+        map.set(key, state.randomValue());
         map.delete(key);
     }
 
+    @Verify(global = true)
```

Comment:
I'm not fine with this verify. There is a loop inside a loop inside a loop. This could lead to a lot of looping at the end of a test.

Reply:
We have more maps - hence the first loop. Each map has more keys - hence the second loop. I don't think this can be don in some simpler way.

The last part:
```
if (!allValues.contains(get)) {
    logger.info(name + ": Expected values: ");
    for (String possibleValue : allValues) {
        logger.info(name + ": Expected value: " + possibleValue);
    }
    logger.info(name + ": Real values: ");
    for (int i = 0; i < keys; i++) {
        logger.info(name + ": Real value: " + mapReference.get(i));
    }
    assertTrue("Value " + get + " for key " + key + " is unexpected.", allValues.contains(get));
}
```
is executed only in case when there is an issue and serves for printing additional information.

If you want to skip `@Verify` then you can use simulator option `verify_enabled: False` for it.

Reply:
Typically this isn't set and it leads to slow verifications on performance tests with large volumes of data.

Reply:
If you want to run "verify" part then I believe it's expected it will do some verification and hence it can take some time - what would be purpose of running verify part without doing verification? If you don't want run verify part then proper option should be set.

I would like to run this verify part in simulator soak tests - we currently plan to run them with `1000` keys and similar amount of values - so not with big data since IIUC CP Map focuses mostly on smaller amount of data, @gbarnett-hz can confirm it I believe.

Again, I can provide some option related to this which will skip verification part (unless it's explicitly allowed) but I don't like it and I believe you don't not like it as well.